ListUtil.xs

```diff
@@ -1445,14 +1474,18 @@ PPCODE:
 
 void
 set_subname(name, sub)
-    char *name
+    SV *name
     SV *sub
 PREINIT:
     CV *cv = NULL;
     GV *gv;
     HV *stash = CopSTASH(PL_curcop);
-    char *s, *end = NULL;
+    const char *s, *end = NULL, *begin = NULL;
     MAGIC *mg;
+    STRLEN namelen;
+    const char* nameptr = SvPV(name, namelen);
+    int utf8flag = SvUTF8(name);
+    int seen_quote = 0, need_subst = 0;
```

Comment:
I'm not terribly comfortable with the bookkeeping of `need_subst` and `seen_quote` added in here.

Both variables are declared as `int` types, with names that feel booleanish, but then one of them is used as a counter anyway.

Can you declare these as

    int quotes_seen = 0;
    bool need_subst = FALSE;

Then the various places that `need_subst++` should become

    need_subst = TRUE;

Reply:
That's probably a historical accident (it's been a while since I touched this, but I think that in an earlier version the two were one variable).